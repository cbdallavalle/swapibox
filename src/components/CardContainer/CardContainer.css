.CardContainer {
  padding: 50px 100px;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.loading-container {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
}

#patience {
  width: 180px;
  padding: 20px 0 0 38px;
}

#vader {
  width: 600px;
  margin: 50px auto;
}
