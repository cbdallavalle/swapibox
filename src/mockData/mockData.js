const mockData = {
  peopleData: {
    name: "Darth Vader",
    homeworld: "Tatooine",
    species: "Human",
    homePop: "200000"

  },
  planetData: {
    name: "Alderaan", 
    terrain: "grasslands, mountains", 
    population: "2000000000", 
    climate: "temperate", 
    residents: [
      {resident: "Leia Organa"},
      {resident: "Bail Prestor Organa"},
      {resident: "Raymus Antilles"}
    ]
  },
  vehicleData: {
    model: "Digger Crawler",
    name: "Sand Crawler",
    passengers: "30",
    vehicleClass: "wheeled"
  },
}

export default mockData;