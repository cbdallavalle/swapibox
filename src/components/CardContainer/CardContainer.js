import React from 'react';
import PropTypes from 'prop-types';
import vader from '../../styles/vader3.jpg';
import patience from '../../styles/patience.gif';
import Card from '../Card/Card';
import './CardContainer.css';

const CardContainer = ({ cardsToRender, toggleTarget }) => {
  const cards = () => {
    if (!cardsToRender.length) {
      return (
        <section className="loading-container">
          <img src={patience} alt="Patience" id="patience" />
          <img src={vader} alt="Patience" id="vader" />
        </section>
      );
    } else {
      return cardsToRender.map( (card, index) => {
        return (
          <Card 
            key={ index }
            card={ card }
            toggleTarget={ toggleTarget }
          />
        );
      });
    }
  };
  
  return (
    <section className="CardContainer">
      { cards() }
    </section>
  );
};

const { object, func, arrayOf } = PropTypes;

CardContainer.propTypes = {
  cardsToRender: arrayOf(object).isRequired,
  toggleTarget: func.isRequired
};

export default CardContainer;