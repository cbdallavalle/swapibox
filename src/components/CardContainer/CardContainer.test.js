/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import CardContainer from './CardContainer.js';
import mockData from '../../mockData/mockData';

describe('CardContainer', () => {
  let wrapper;
  const mockToggleTarget = jest.fn();

  it('should match snapshot when there are no cards to render', () => {
    wrapper = shallow(          
      <CardContainer cardsToRender={ [] }
                     toggleTarget={ mockToggleTarget }
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should match snapshot when there are cards to render', () => {
    wrapper = shallow(
      <CardContainer cardsToRender={ [ mockData.peopleData ] }
                     toggleTarget={ mockToggleTarget }
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
