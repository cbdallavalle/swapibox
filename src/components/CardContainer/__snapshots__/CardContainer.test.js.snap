// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match snapshot when there are cards to render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    cardsToRender={
        Array [
            Object {
              "homePop": "200000",
              "homeworld": "Tatooine",
              "name": "Darth Vader",
              "species": "Human",
            },
          ]
    }
    toggleTarget={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          card={
                    Object {
                              "homePop": "200000",
                              "homeworld": "Tatooine",
                              "name": "Darth Vader",
                              "species": "Human",
                            }
          }
          toggleTarget={[Function]}
/>,
      ],
      "className": "CardContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "card": Object {
            "homePop": "200000",
            "homeworld": "Tatooine",
            "name": "Darth Vader",
            "species": "Human",
          },
          "toggleTarget": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            card={
                        Object {
                                    "homePop": "200000",
                                    "homeworld": "Tatooine",
                                    "name": "Darth Vader",
                                    "species": "Human",
                                  }
            }
            toggleTarget={[Function]}
/>,
        ],
        "className": "CardContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "card": Object {
              "homePop": "200000",
              "homeworld": "Tatooine",
              "name": "Darth Vader",
              "species": "Human",
            },
            "toggleTarget": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CardContainer should match snapshot when there are no cards to render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    cardsToRender={Array []}
    toggleTarget={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="loading-container"
>
        <img
                alt="Patience"
                id="patience"
                src="patience.gif"
        />
        <img
                alt="Patience"
                id="vader"
                src="vader3.jpg"
        />
</section>,
      "className": "CardContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Patience"
            id="patience"
            src="patience.gif"
/>,
          <img
            alt="Patience"
            id="vader"
            src="vader3.jpg"
/>,
        ],
        "className": "loading-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Patience",
            "id": "patience",
            "src": "patience.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Patience",
            "id": "vader",
            "src": "vader3.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "section",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="loading-container"
>
          <img
                    alt="Patience"
                    id="patience"
                    src="patience.gif"
          />
          <img
                    alt="Patience"
                    id="vader"
                    src="vader3.jpg"
          />
</section>,
        "className": "CardContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Patience"
              id="patience"
              src="patience.gif"
/>,
            <img
              alt="Patience"
              id="vader"
              src="vader3.jpg"
/>,
          ],
          "className": "loading-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Patience",
              "id": "patience",
              "src": "patience.gif",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Patience",
              "id": "vader",
              "src": "vader3.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "section",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
